pjmedia/include/pjmedia/txt_stream.h

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (C) 2024-2025 Teluu Inc. (http://www.teluu.com)
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#ifndef __PJMEDIA_TXT_STREAM_H__
+#define __PJMEDIA_TXT_STREAM_H__
+
+
+/**
+ * @file txt_stream.h
+ * @brief Text Stream.
+ */
+
+#include <pjmedia/endpoint.h>
+#include <pjmedia/jbuf.h>
+#include <pjmedia/port.h>
+#include <pjmedia/rtcp.h>
+#include <pjmedia/rtcp_fb.h>
+#include <pjmedia/transport.h>
+#include <pjmedia/stream_common.h>
+
+PJ_BEGIN_DECL
+
+
+/**
+ * @defgroup PJMED_TXT_STRM Text streams
+ * @ingroup PJMEDIA_PORT
+ * @brief Text communication via the network
+ * @{
+ *
+ * A text stream is a bidirectional text communication between two
+ * endpoints. It corresponds to a text media description ("m=text" line)
+ * in SDP session descriptor.
+ *
+ * A text stream consists of two unidirectional channels:
+ *  - encoding channel, which transmits unidirectional text to remote, and
+ *  - decoding channel, which receives unidirectional text from remote.
+ *
+ * A text stream internally manages the following objects:
+ *  - a @ref PJMED_JBUF,
+ *  - two instances of RTP sessions (#pjmedia_rtp_session, one for each
+ *    direction),
+ *  - one instance of RTCP session (#pjmedia_rtcp_session),
+ *  - and a reference to media transport to send and receive packets
+ *    to/from the network (see @ref PJMEDIA_TRANSPORT).
+ *
+ * Text streams are created by calling #pjmedia_txt_stream_create(),
+ * specifying #pjmedia_txt_stream_info structure in the parameter. Application
+ * can construct the #pjmedia_txt_stream_info structure manually, or use 
+ * #pjmedia_txt_stream_info_from_sdp() function to construct the
+ * #pjmedia_txt_stream_info from local and remote SDP session descriptors.
```

Comment:
Should it also mentions the RFC the implementation is based on, so users get a picture of what to expect (features, interoperability, etc)